tls-core/src/ghash/mod.rs

```diff
@@ -0,0 +1,295 @@
+//! ghash implements a protocol of computing the AES-GCM's GHASH function in a
+//! secure two-party computation (2PC) setting using 1-out-of-2 Oblivious
+//! Transfer (OT). The parties start with their secret XOR shares of H (the
+//! GHASH key) and at the end each gets their XOR share of the GHASH output.
+//! The method is decribed here:
+//! (https://tlsnotary.org/how_it_works#section4).
+
+//! As an illustration, let's say that S has his shares H1_s and H2_s and R
+//! has her shares H1_r and H2_r. They need to compute shares of H3.
+//! H3 = (H1_s + H1_r)*(H2_s + H2_r) = H1_s*H2_s + H1_s*H2_r + H1_r*H2_s +
+//! H1_r*H2_r. Term 1 can be computed by S locally and term 4 can be
+//! computed by R locally. Only terms 2 and 3 will be computed using
+//! GHASH 2PC. R will obliviously request values for bits of H1_r and H2_r.
+//! The XOR sum of all values which S will send back plus H1_r*H2_r will
+//! become R's share of H3.
+//!
+//! When performing block multiplication in 2PC, Master holds the Y value and
+//! Slave holds the X value. The Slave then computes the X table and masks it.
+
+mod common;
+mod errors;
```

Comment:
The errors module should be public so users can import and design their control flow around them